monai/transforms/compose.py

```diff
@@ -21,21 +21,64 @@
 import numpy as np
 
 import monai
+import monai.transforms as mt
 from monai.transforms.inverse import InvertibleTransform
 
 # For backwards compatibility (so this still works: from monai.transforms.compose import MapTransform)
 from monai.transforms.transform import (  # noqa: F401
+    LazyTransform,
     MapTransform,
     Randomizable,
     RandomizableTransform,
     Transform,
     apply_transform,
 )
-from monai.utils import MAX_SEED, ensure_tuple, get_seed
-from monai.utils.enums import TraceKeys
+from monai.utils import MAX_SEED, TraceKeys, ensure_tuple, get_seed
 
 __all__ = ["Compose", "OneOf", "RandomOrder"]
 
+from monai.utils.misc import to_tuple_of_dictionaries
+
+
+def _evaluate_with_overrides(
+    data,
+    upcoming,
+    lazy_evaluation: bool | None = False,
+    overrides: dict | None = None,
+    override_keys: Sequence[str] | None = None,
+):
+    """
+    Given the upcoming transform ``upcoming``, if lazy_resample is True, go through the MetaTensors and
+    evaluate the lazy applied operations. The returned `data` will then be ready for the ``upcoming`` transform.
+    """
+    if not lazy_evaluation:
+        return data  # eager evaluation
+    overrides = (overrides or {}).copy()
+    if isinstance(data, monai.data.MetaTensor):
+        if data.has_pending_operations() and ((isinstance(upcoming, (mt.Identityd, mt.Identity))) or upcoming is None):
+            device = overrides.pop("device", None)
+            if device is not None:
+                data = mt.EnsureType(device=device)(data)
+            data, _ = mt.apply_transforms(data, None, overrides=overrides)
+        return data
+    override_keys = ensure_tuple(override_keys)
+    if isinstance(data, dict):
+        if isinstance(upcoming, MapTransform):
+            _keys = [k if k in upcoming.keys and k in data else None for k in override_keys]  # type: ignore
```

Comment:
when override_keys are provided but don't match any of the transform/data keys, this should have a warning message

Reply:
I agree, a warning message can improve the user experience here.

Thanks.

Reply:
Hmmm. Is it actually a warning? You might omit a key because you don't need to specify an override for them. You might need to change ```mode``` for 'image' if it is set to one of the various ```bilinear```-style keys, but ```label``` might already be set to ```nearest```.

Reply:
sure, in that logic, empty `override_keys` means check all the data inputs for lazy evaluation, I'll change. previously this variable was `lazy_key`, which means if it's empty, then no input is checked for lazy evaluation. I'll think about it.